File: pages/main_page.py
from .base_page import BasePage
from selenium.webdriver.common.by import By
from pages.blog_page import BlogPage


class MainPage(BasePage):
    # Page url
    main_page_url = "https://distillery.com/"

    # Page Locators
    header_locator = (By.CSS_SELECTOR, ".jumbotron-home__header")
    learn_about_us_button = (By.CSS_SELECTOR, "#focus-features-btn [href='https://distillery.com/about/']")
    blog_button = (By.CSS_SELECTOR, ".navigation-menu__container [data-number='7']")

    # Page info
    main_page_header_text = "DISTILLERY IS A\nFULL−SERVICE SOFTWARE\nDESIGN AND DEVELOPMENT COMPANY"

    def __init__(self, *args, **kwargs):
        super(MainPage, self).__init__(*args, **kwargs)

    def open(self):
        self.navigate_to(self.main_page_url)
        return self

    def open_blog(self):
        self.get_element(self.blog_button).click()
        return BlogPage(self.driver)

File: src/test/java/pages/MainPage.java
package pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public class MainPage extends BasePage {
    private final String HOME_PAGE_LABEL = "xpath://h1[@class='jumbotron-home__header']";
    private final String LEARN_ABOUT_US_BUTTON = "id:focus-features-btn";

    public MainPage(WebDriver driver) {
        super(driver);
    }

    public void openMainPage() {
        driver.get("https://distillery.com/");
    }

    public String getPageTitle() {
        WebElement element = waitForElementPresent(HOME_PAGE_LABEL, "Cannot get home page label!", 10);
        return element.getText();
    }

    public WebElement getLearnAboutUsButton() {
        return waitForElementPresent(LEARN_ABOUT_US_BUTTON, "Cannot locate 'Learn About Us' button!", 10);
    }
}
